tests/visualization_tests/test_terminator_improvement.py

```diff
@@ -0,0 +1,124 @@
+from optuna.terminator import report_cross_validation_scores
+from optuna.terminator import StaticErrorEvaluator
+from optuna.testing.objectives import fail_objective
+from optuna.testing.visualization import prepare_study_with_trials
+from optuna.visualization import plot_terminator_improvement as plotly_plot_terminator_improvement
+from optuna.visualization._terminator_improvement import _get_improvement_info
+from optuna.visualization._terminator_improvement import _get_y_range
+from optuna.visualization._terminator_improvement import _ImprovementInfo
+
+
+parametrize_plot_terminator_improvement = pytest.mark.parametrize(
+    "plot_terminator_improvement", [plotly_plot_terminator_improvement]
+)
+
+
+def _create_study_with_failed_trial() -> Study:
+    study = create_study()
+    study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
+    return study
+
+
+def test_study_is_multi_obj() -> None:
+    study = create_study(directions=["minimize", "minimize"])
+    with pytest.raises(ValueError):
+        _get_improvement_info(study=study)
+
+
+@parametrize_plot_terminator_improvement
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [
+        create_study,
+        _create_study_with_failed_trial,
+        prepare_study_with_trials,
+    ],
+)
+@pytest.mark.parametrize("plot_error", [False, True])
+def test_plot_terminator_improvement(
+    plot_terminator_improvement: Callable[..., Any],
+    specific_create_study: Callable[[], Study],
+    plot_error: bool,
+) -> None:
+    study = specific_create_study()
+    figure = plot_terminator_improvement(study)
+    figure.write_image(BytesIO())
+
+
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [create_study, _create_study_with_failed_trial],
+)
+@pytest.mark.parametrize("plot_error", [False, True])
+def test_get_terminator_improvement_info_empty(
+    specific_create_study: Callable[[], Study], plot_error: bool
+) -> None:
+    study = specific_create_study()
+    info = _get_improvement_info(study, plot_error)
+    assert info == _ImprovementInfo(trial_numbers=[], improvements=[], errors=None)
+
+
+@pytest.mark.parametrize("get_error", [False, True])
+@pytest.mark.parametrize("improvement_evaluator_class", [lambda: RegretBoundEvaluator(), None])
+@pytest.mark.parametrize(
+    "error_evaluator_class",
+    [
+        lambda: CrossValidationErrorEvaluator(),
+        lambda: StaticErrorEvaluator(0),
+        None,
+    ],
+)
+def test_get_improvement_info(
+    get_error: bool,
+    improvement_evaluator_class: Callable[[], BaseImprovementEvaluator] | None,
+    error_evaluator_class: Callable[[], BaseErrorEvaluator] | None,
+) -> None:
+    study = create_study()
+    for _ in range(3):
+        trial = study.ask({"x": FloatDistribution(0, 1)})
+        report_cross_validation_scores(trial, [1.0, 2.0])
+        study.tell(trial, 0)
+
+    improvement_evaluator = (
+        None if improvement_evaluator_class is None else improvement_evaluator_class()
+    )
```

Comment:
If we replace `None` with `lambda: None` at the corresponding parametrization, we may be able to simplify these lines.